### PR TITLE
ovl_Efc_Erupc OK

### PR DESCRIPTION
**Pull request by mzxrules**
also pretty easy

### CODE REVIEW

File: src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c [hunk 1]
```diff
@@ -26,23 +31,230 @@ const ActorInit Efc_Erupc_InitVars = {
     (ActorFunc)EfcErupc_Update,
     (ActorFunc)EfcErupc_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Efc_Erupc/func_8099CCB0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Efc_Erupc/EfcErupc_Init.s")
+void EfcErupc_SetupAction(EfcErupc* this, EfcErupcActionFunc actionFunc) {
+
```
Comment (louist103) [inline src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c:36]:
```suggestion
```

File: src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c [hunk 2]
```diff
@@ -26,23 +31,230 @@ const ActorInit Efc_Erupc_InitVars = {
...
+    this->actionFunc = actionFunc;
+}
+
+void EfcErupc_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EfcErupc* this = THIS;
+
+    EfcErupc_SetupAction(this, &EfcErupc_UpdateAction);
+    Actor_SetScale(thisx, 1.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c:44]:
```suggestion
    Actor_SetScale(&this->actor, 1.0f);
```

File: src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c [hunk 3]
```diff
@@ -26,23 +31,230 @@ const ActorInit Efc_Erupc_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Efc_Erupc/func_8099CD2C.s")
+    func_80093D84(globalCtx->state.gfxCtx);
+
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, this->unk14C * 1, this->unk14E * -4, 32, 64, 1,
+                                this->unk14C * 4, this->unk14E * -20, 64, 64));
+
+    gSPSegment(
+        POLY_XLU_DISP++, 0x09,
+        Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0U, this->unk150 * -4, 16, 128, 1, 0, this->unk150 * 12, 32, 32));
```
Comment (louist103) [inline src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c:134]: Are the `U`s required?

File: src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c [hunk 4]
```diff
@@ -26,23 +31,230 @@ const ActorInit Efc_Erupc_InitVars = {
...
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_efc_erupc.c", 356);
+    EfcErupc_DrawParticles(this->particles, globalCtx);
+}
+
+void EfcErupc_DrawParticles(EfcErupcParticles* particles, GlobalContext* globalCtx) {
+    GraphicsContext* gfxCtx;
+    s16 i;
+    s32 pad;
+
+    gfxCtx = globalCtx->state.gfxCtx;
```
Comment (louist103) [inline src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c:174]:
```suggestion
    GraphicsContext*  gfxCtx = globalCtx->state.gfxCtx;;
    s16 i;
    s32 pad;
    
```

File: src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c [hunk 5]
```diff
@@ -26,23 +31,230 @@ const ActorInit Efc_Erupc_InitVars = {
...
+            Matrix_Scale(particles->scale, particles->scale, 1.0f, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx, "../z_efc_erupc.c", 393),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, D_060027D8);
+        }
+    }
+    CLOSE_DISPS(gfxCtx, "../z_efc_erupc.c", 399);
+}
+
+Color_RGB8 D_8099D770[] = {
```
Comment (louist103) [inline src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c:195]:
```suggestion
static Color_RGB8 D_8099D770[] = {
```

File: src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.h [hunk 6]
```diff
@@ -6,9 +6,32 @@
 
 struct EfcErupc;
 
+typedef void (*EfcErupcActionFunc)(struct EfcErupc*, GlobalContext*);
+
+typedef struct {
```
Comment (louist103) [inline src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.h:11]: Offsets like in the actor struct.

File: src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c [hunk 7]
```diff
@@ -26,23 +31,230 @@ const ActorInit Efc_Erupc_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Efc_Erupc/func_8099D52C.s")
+    for (i = 0; i < EFC_ERUPC_NUM_PARTICLES; i++, particles++) {
+        if (!particles->isActive) {
+            particles->isActive = true;
+            particles->pos = *pos;
+            particles->vel = *vel;
+            particles->accel = *accel;
+            particles->scale = scaleFactor / 1000.0f;
+            particles->alpha = 255;
+            particles->animTimer = (s8)(s32)(s8)(s32)Rand_ZeroFloat(10);
```
Comment (louist103) [inline src/overlays/actors/ovl_Efc_Erupc/z_efc_erupc.c:248]: This seems really fake but until someone has a better idea I guess its OK.  Maybe leave a note though.

**Review by louist103 — CHANGES_REQUESTED** [review]
Looks good.  Just some small things.

**Review by fig02 — APPROVED** [review]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**